vital/types/activity_type.h

```diff
@@ -0,0 +1,280 @@
+/*ckwg +30
+ * Copyright 2020 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be
+ *    used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief Header for an activity type
+ */
+
+#ifndef VITAL_TYPES_ACTIVITY_TYPE_H_
+#define VITAL_TYPES_ACTIVITY_TYPE_H_
+
+#include <vital/vital_export.h>
+#include <vital/vital_types.h>
+
+#include <vector>
+#include <map>
+#include <set>
+#include <memory>
+#include <utility>
+
+namespace kwiver {
+namespace vital {
+
+// ----------------------------------------------------------------
+/**
+ * @brief Activity type classification.
+ *
+ * This class represents a set of possible types for the activity.
+ * A type for an activity is represented by a class_name string
+ * and a score.
+ *
+ * When an activity is classified, there may be several possibilities
+ * determined, since the classification process is probabilistic. This
+ * class captures the set of possible types along with the relative
+ * likelihood or score.
+ *
+ * Note that score values in are *not* constrained to
+ * [0.0,1.0] because different detectors use different approaches for
+ * scores. These scores can be normalized, but that is up to the user
+ * of these values.
+ */
+
+class VITAL_EXPORT activity_type
```

Comment:
The `activity_type` class looks very similar to the `detected_object_type` class. We could make a more generic class that encapsulates a class-label & score and use it in both places. Also may be useful in additional places in the future.

Reply:
I agree that a more generic class with class label and scores would be useful. Should it be added in this branch?

Reply:
Just to be clear, does this mean an abstract parent class with virtual methods implemented by `activity_type` and `detected_object_type`?

Reply:
These changes are in #1018